common/frame_manager.go

```diff
@@ -592,10 +592,23 @@ func (m *FrameManager) NavigateFrame(frame *Frame, url string, parsedOpts *Frame
 	lifecycleEvtCh, lifecycleEvtCancel := createWaitForEventPredicateHandler(
 		timeoutCtx, frame, []string{EventFrameAddLifecycle},
 		func(data any) bool {
-			if le, ok := data.(LifecycleEvent); ok {
-				return le == parsedOpts.WaitUntil
+			le, ok := data.(FrameLifecycleEvent)
+			if !ok {
+				return false
 			}
-			return false
+			// skip the initial blank page if we are navigating to a non-blank page.
+			// otherwise, we will get a lifecycle event for the initial blank page
+			// and return prematurely before waiting for the navigation to complete.
+			if url != BlankPage && le.URL == BlankPage {
+				m.logger.Debugf(
+					"FrameManager:NavigateFrame:createWaitForEventPredicateHandler",
+					"fmid:%d fid:%v furl:%s url:%s waitUntil:%s event.lifecycle:%q event.url:%q skipping %s",
+					fmid, fid, furl, url, parsedOpts.WaitUntil, le.Event, le.URL, BlankPage,
+				)
+				return false
```

Comment:
Thanks for working on this @inancgumus !

I think this change makes complete sense, but I'm concerned because in practice I don't see a difference in the Web Vitals metrics being reported. For example, with the following test:
<details>
<summary>Simple test example.</summary>

```js
import { browser } from 'k6/x/browser';

export const options = {
  scenarios: {
    ui: {
      executor: 'shared-iterations',
      options: {
        browser: {
            type: 'chromium',
        },
      },
    },
  },
};

export default async function () {
  const page = browser.newPage();

  try {
    await page.goto('https://test.k6.io/')
  } finally {
    page.close();
  } 
}
```

</details>

I only see the `FCP` and `TTFB` being reported, but I don't see the `CLS` and `LCP` metrics, which I believe should always be reported after a navigation. And in this sense this is the same behavior we have in `main` right now. Did you observe something different? Am I doing or understanding something wrong? :thinking: 

In any case I understand the change is a step forward on improving this! :clap:

Reply:
Sure thing ❤️ 

I don't see a difference, too... It was flaky when I started working on this, but now it constantly works on `main`. That's super strange! The only moving parts are the `test.k6.io` site and my network connection. There were network problems a few days back. Maybe the bug only surfaces in those cases. I'm still determining.

I noticed `CLS` and `LSP` are only fired if we make another navigation after the initial one: `page.goto` then `page.reload` or similar. Are we sure that they only get fired for the initial navigation?

When I look into the trace logs, I see that:
1. `NewBrowserContext` adds the Web Vital JSes (`web_vital_[iife|init].js`).
2. Then `NewPage` registers the bindings.
3. The initial frame navigation happens.
4. `beforeunload` and `pagehide` events happen. Usually, as I explained above, after the initial navigation, these events trigger the `CLS` and `LSP` metrics. But, for the initial navigation, they do not.

What this PR does is wait for the navigation to end prematurely, and it's a good first step towards fixing this general Web Vitals issue 🤞

Reply:
@ka3de, also see [this fix](https://github.com/grafana/xk6-browser/pull/949).